Unix/http/httpclient.c

```diff
@@ -2566,32 +2566,43 @@ MI_Result HttpClient_New_Connector2(
         // Parse the user name to separate the domain/host from the username. This is either in the form
         // domain\user or user@domain or just user. mit krb5 can handle either, but heimdal doesn't.
         // Once we have the domain sorted out we can use it later on in generating the target name so
-        // it isn't a waste either way
+        // it isn't a waste either way. Basic auth doesn't need parsing because PAM does the work there.
 
         if (username)
-        {
-            user_domain = memchr(username, '\\', strlen(username));
-            if (user_domain)
+        { 
+
+            if ( authtype == AUTH_METHOD_BASIC)
             {
-                // The form would be "domain\user'
-                *user_domain = '\0';
-                self->connector->username    = PAL_Strdup(user_domain+1);
-                self->connector->user_domain = username;
+                self->connector->username    = username;
+                self->connector->user_domain = NULL;
             }
-            else
+            else if ( authtype == AUTH_METHOD_NEGOTIATE ||
+                      authtype == AUTH_METHOD_NEGOTIATE_WITH_CREDS ||
+                      authtype == AUTH_METHOD_KERBEROS) 
             {
-                user_domain = memchr(username, '@', strlen(username));
+                user_domain = memchr(username, '\\', strlen(username));
```

Comment:
strchr is probably quicker as your solution always searches for the null terminator where strchr looks for the char until it hits the null terminator.